Source/WebCore/accessibility/AccessibilityRenderObject.cpp

```diff
@@ -1078,6 +1078,26 @@ static bool webAreaIsPresentational(RenderObject* renderer)
     return ownerElement && nodeHasPresentationRole(*ownerElement);
 }
 
+bool AccessibilityRenderObject::shouldWebAreaExposeChildren() const
+{
+    if (!isWebArea())
+        return true;
+
+    CheckedPtr renderView = dynamicDowncast<RenderView>(m_renderer.get());
+    CheckedPtr frameRenderer = renderView ? renderView->frameView().frame().ownerRenderer() : nullptr;
+
+    while (frameRenderer) {
+        const auto usedVisibility = frameRenderer->style().usedVisibility();
+        if (usedVisibility != Visibility::Visible || frameRenderer->style().effectiveInert())
+            return false;
+
+        renderView = frameRenderer->document().renderView();
+        frameRenderer = renderView ? renderView->frameView().frame().ownerRenderer() : nullptr;
+    }
+
```

Comment:
Take it or leave it: This newline feels unnecessary to me, but maybe y'all have other preferences